### ISSUE TITLE
Improve Logging

### ISSUE THREAD

**Issue by JerryNixon** [Windows-XAML/Template10#1664]
I am pretty satisfied with the changes to logging I have made.

- [x] Create default LoggerFacade (uses Debug.WriteLine())
- [x] Made method starting logs optional
- [ ] Find a way to turn on method starting logs later
- [x] Change all the logging strings to make more sense
- [x] Update the application base to use logging as soon as it is available
- [x] In application base don't throw without logging, first
- [x] In navigation service don't throw without logging, first
- [x] In frame façade don't throw without logging, first

![image](https://user-images.githubusercontent.com/1749983/45602194-ecb19100-b9d6-11e8-91b0-66f60f2f5c4a.png)

**Comment by JerryNixon** [comment]
I think it is interesting and perhaps not immediately intuitive that `App.RegisterTypes()` is called in the application before the `App.Constructor()` is called. This might even seem impossible, but, of course it is not. This is because `RegisterTypes()` is called from `base.Constructor()` and means the container is finalized before the application really starts, and the container can resolve immediately in the application's constructor. It's by design, just pointing it out.

**Comment by JerryNixon** [comment]
I made all Info/None logs not display Info/Log in the preamble. Because of this, the logs are easier to read and Template 10 line items are mostly indented. I have made navigation events like Nav, Back, Forward, and Refresh all Info/Low so they contain the prefix and stick out in the output - this makes it slightly easier to glance at the output window and trace with your eyes what is happening and when events that are more important occur.
![image](https://user-images.githubusercontent.com/1749983/45602251-2931bc80-b9d8-11e8-9594-eb925a9706f8.png)
